src/vehicle_move.cpp

```diff
@@ -1071,8 +1071,8 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
             }
         }
 
-        if( critter == nullptr || !critter->is_hallucination() ) {
-            coll_velocity = vel1_a * ( smashed ? 100 : 90 );
+        if( critter == nullptr || ( critter != nullptr && !critter->is_hallucination() ) ) {
```

Comment:
```suggestion
        if( critter == nullptr || !critter->is_hallucination() ) {
```

The condition will already return true if `critter` is a nullptr, so it will never be a nullptr when checking if it's a hallucination. Checking it again is unnecessary.